CMakeLists.txt

```diff
@@ -763,6 +763,10 @@ if(HPX_COMPILER_WARNINGS)
     if(MPI_FOUND)
       hpx_add_compile_flag(-wd4800)
     endif()
+
+    hpx_add_compile_flag(-wd4800)
```

Comment:
This warning is conditionally disabled in the lines above, is that code still needed? If this warning comes from the new code, could it just be silenced in the normal way, by explicitly converting to `bool` one way or another?